src/tasks/task1.js

```diff
@@ -0,0 +1,57 @@
+/* TODO: #1. Функция принимает массив пользователей. Возвращает объект, состоящий из двух полей: women и men. Данные
+поля являются массивами, в которых содержаться пользователи, подходящие по категории gender соответственно.
+Также, вместо двух полей first_name и last_name у каждого из объектов должнобыть поле fullName в котором будут объеденины убранные поля
+(first_name и last_name). Количество пользователей может быть не ограничено. */
+
+const users = [
+	{
+		id: 1,
+		first_name: 'Jeanette',
+		last_name: 'Penddreth',
+		email: 'jpenddreth0@census.gov',
+		gender: 'Female',
+		ip_address: '26.58.193.2',
+	},
+	{
+		id: 2,
+		first_name: 'Petr',
+		last_name: 'Stepanov',
+		email: 'gfrediani1@senate.gov',
+		gender: 'Male',
+		ip_address: '229.179.4.212',
+	},
+	{
+		id: 3,
+		first_name: 'Michael',
+		last_name: 'Jackson',
+		email: 'gfrediani1@senate.gov',
+		gender: 'Male',
+		ip_address: '229.179.4.212',
+	},
+];
+
+
+const sortByGender = (usersArray) => {
+	if (!Array.isArray(usersArray)) return;
+	const result = {};
+	result['women'] = [];
+	result['men'] = [];
+	copiedArray = [...usersArray];
+	copiedArray.forEach(user => {
+		if (user.hasOwnProperty('first_name') && user.hasOwnProperty('last_name') && user.hasOwnProperty('gender')) {
+			user['fullName'] = `${user.first_name} ${user.last_name}`;
+			delete user.first_name;
+			delete user.last_name;
+			if (user.gender === "Female") {
+				result['women'].push(user);
+			} else {
+				result['men'].push(user);
+			}
+		}
+	})
+
+	return result;
+}
+
+
```

Comment:
The most perfect and nice way to implement this task is to use reduce() method. Thry to do this with it.